bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -134,6 +138,21 @@ private void refreshPlayerState() {
         synchronized (sequenceLock) {
             String activeAppId = ROKU_HOME_ID;
             try {
+                if (thingTypeUID.equals(THING_TYPE_ROKU_TV)) {
+                    try {
+                        deviceInfo = communicator.getDeviceInfo();
+                        String powerMode = deviceInfo.getPowerMode();
+                        updateState(POWER_STATE, new StringType(powerMode));
+                        if (POWERON.equals(powerMode)) {
+                            updateState(POWER, OnOffType.ON);
+                        } else {
+                            updateState(POWER, OnOffType.OFF);
```

Comment:
If the TV is off you could return here to avoid all un-necessary queries below.
```suggestion
                            updateState(POWER, OnOffType.OFF);
                            return;
```

Reply:
I'll add it.